base_layer/wallet_ffi/src/lib.rs

```diff
@@ -2331,6 +2370,58 @@ pub unsafe extern "C" fn comms_config_create(
     }
 }
 
+/// Set the Comms Secret Key for an existing TariCommsConfig. Usually this key is maintained by the backend but if it is
+/// required to set a specific new one this function can be used.
+///
+/// ## Arguments
+/// `comms_config` - TariCommsConfig to be updated
+/// `secret_key` - The TariSecretKey pointer. This is the secret key corresponding to the Public key that represents
+/// this node on the Tari comms network
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// None
+///
+/// # Safety
+/// None
+#[no_mangle]
+pub unsafe extern "C" fn comms_config_set_secret_key(
```

Comment:
```
pub unsafe extern "C" fn comms_config_set_secret_key(
    comms_config: *const TariCommsConfig,
    secret_key: *const TariPrivateKey,
    error_out: *mut c_int,
)
```

Reply:
no the config needs to be mut because this function modifies it.

Reply:
Ooops! Meant `secret_key` should be `*const`

Reply:
done